### DISCUSSION TITLE
怎么让控制台使用公网

### DISCUSSION THREAD

**Discussion by DYMTHH** [koishijs/koishi#973] [Q&A]
### Describe the problem related to the feature request

怎么让控制台使用公网

### Describe the solution you'd like

我用的宝塔，liunx系统，怎么让他不用内网，用公网

**Comment by Lipraty** [comment]
你好请前往论坛查看你需要的内容：https://forum.koishi.xyz/t/topic/294

---
Generated by [koishi-plugin-github](https://github.koishi.chat/).

**Comment by DYMTHH** [comment]
> 里面说到进入控制台修改IP，但是我现在无法进入控制台



> 你好请前往论坛查看你需要的内容：https://forum.koishi.xyz/t/topic/294
> 
> 由[koishi-plugin-github](https://github.koishi.chat/)生成。

里面说到进入控制台修改IP，但是我现在无法进入控制台

**Comment by Lipraty** [comment]
> > 里面说到进入控制台修改IP，但是我现在无法进入控制台
> 
> > 你好请前往论坛查看你需要的内容：https://forum.koishi.xyz/t/topic/294
> > 由[koishi-plugin-github](https://github.koishi.chat/)生成。
> 
> 里面说到进入控制台修改IP，但是我现在无法进入控制台

那你可以停止 koishi-desktop 后结合 [这篇](https://forum.koishi.xyz/t/topic/288) 帖子，在 koishi.yml 中修改地址为 `0.0.0.0`

**Comment by DYMTHH** [comment]
> > > 里面说到进入控制台修改IP，但是我现在无法进入控制台
> > 
> > 
> > > 你好请前往论坛查看你需要的内容：https://forum.koishi.xyz/t/topic/294
> > > 由[koishi-plugin-github](https://github.koishi.chat/)生成。
> > 
> > 
> > 里面说到进入控制台修改IP，但是我现在无法进入控制台
> 
> 那你可以停止 koishi-desktop 后结合 [这篇](https://forum.koishi.xyz/t/topic/288) 帖子，在 koishi.yml 中修改地址为`0.0.0.0`

宝塔的Docker没有编辑文件的能力（（（（（（（

**Comment by Lipraty** [comment]
> > > > 里面说到进入控制台修改IP，但是我现在无法进入控制台
> > > 
> > > 
> > > > 你好请前往论坛查看你需要的内容：https://forum.koishi.xyz/t/topic/294
> > > > 由[koishi-plugin-github](https://github.koishi.chat/)生成。
> > > 
> > > 
> > > 里面说到进入控制台修改IP，但是我现在无法进入控制台
> > 
> > 
> > 那你可以停止 koishi-desktop 后结合 [这篇](https://forum.koishi.xyz/t/topic/288) 帖子，在 koishi.yml 中修改地址为`0.0.0.0`
> 
> 宝塔的Docker没有编辑文件的能力（（（（（（（

你在使用 Docker 时应当阅读 [官方文档](https://koishi.chat/manual/starter/docker.html) ，并看到了第一行 warning 所述内容：

> Docker 是一个以服务生产环境而开发的应用平台，在使用 Docker 部署之时，我们相信你已经掌握了运维一台服务器所必须的知识，同时也理解了容器化的概念与 Docker 的基础操作。如若不然，在除路由器或 NAS 等特殊环境外，请 [选择其他安装方式](https://koishi.chat/manual/starter/)。

**Comment by H4M5TER** [comment]
你是用什么方法启动的 docker 呢，如果是官方的 docker 镜像的话，已经是使用公网的了，是否你没有进行端口映射或者没有正确配置好防火墙呢

**Comment by DYMTHH** [comment]
> 

宝塔，腾讯服务器宝塔liunx

**Comment by H4M5TER** [comment]
你是否理解我的意思
我的意思是：如果你是用面板拉取并启动了官方的镜像的话，你不需要编辑文件改变 host，问题出在别的地方

**Reply by DYMTHH** [reply]
显示的是http://[127.0.0.1](http://127.0.0.1:5140/):5140/ ，使用公网IP无法访问，端口已经开放了

**Reply by DYMTHH** [reply]
> 你是否理解我的意思 我的意思是：如果你是用面板拉取并启动了官方的镜像的话，你不需要编辑文件改变 host，问题出在别的地方

我指的是服务器上面显示的IP，不是本地

**Reply by MaikoTan** [reply]
Can you post any screenshot rather than recycling the same words?

**Reply by DYMTHH** [reply]
so how I send you screenshot

**Reply by MaikoTan** [reply]
Just drag and drop your screenshot into the text box on github, would be automatically uploaded.

**Reply by DYMTHH** [reply]
l find other workaround thanks

**Comment by MaikoTan** [comment]
This is not a feature request, I would move it to discussion and close this one.

**Comment by acgurl** [comment]
我想知道修改koishi.yml中的哪一条配置为0.0.0.0
我的配置文件如下
```
port: 5140
maxPort: 5149

plugins:
  group:basic:
    ~admin: {}
    ~bind: {}
    commands: {}
    help: {}
    ~inspect: {}
    locales: {}
    rate-limit: {}
  group:console:
    analytics: {}
    ~auth: {}
    config: {}
    console:
      open: true
    dataview: {}
    desktop:
      $if: env.KOISHI_AGENT?.includes('Desktop')
    explorer: {}
    logger: {}
    insight: {}
    market:
      registry:
        endpoint: https://registry.npmmirror.com
      search:
        endpoint: https://registry.koishi.chat/index.json
    sandbox: {}
    status: {}
    theme-vanilla: {}
  group:storage:
    database-sqlite: {}
    ~database-mysql: {}
    ~database-mongo: {}
    assets-local: {}
  group:adapter:
    ~adapter-onebot: {}
    ~adapter-discord: {}
    ~adapter-telegram: {}
    ~adapter-kook: {}
    ~adapter-lark: {}
    gocqhttp:
      signServer: https://captcha.koishi.xyz/sdk/dandelion/energy
  group:develop:
    $if: env.NODE_ENV === 'development'
    hmr:
      root: '.'
```

**Reply by ilharp** [reply]
上方的讨论已经解决了此问题。请仔细阅读上方的讨论。

**Reply by acgurl** [reply]
我的意思是填上这个0.0.0.0？

**Reply by ilharp** [reply]
上方的讨论已经解决了此问题。请仔细阅读上方讨论中给出的解决方法，而不是就相同的问题再次提问。

**Reply by shigma** [reply]
如果你会编辑 yaml 那就加一行 host: 0.0.0.0，也可以阅读上方讨论的解法。

**Reply by acgurl** [reply]
好的，非常感谢
